### DISCUSSION TITLE
Community toolkit components and dependency injection

### DISCUSSION THREAD

**Discussion by durandt** [CommunityToolkit/Maui#976] [General]
Context: I am working on a BottomSheet ( #143 ) component

What are the current recommended guidelines for dependency injection for community toolkit components? 
If my component needs to use (for example) `IDeviceDisplay`, should I add `services.AddSingleton<IDeviceDisplay>(DeviceDisplay.Current);` to the `UseMauiCommunityToolkit` extension-method, or should I add IDeviceDisplay in the constructor of my component and let it up to the dev/project using the component to resolve IDeviceDisplay?

**Comment by VladislavAntonyuk** [comment]
Before working on it, the discussion should be converted to Feature and approved (see https://github.com/CommunityToolkit/Maui#submitting-a-new-feature)
You can use `DeviceDisplay.Current` directly without needing to use DI.
it will help avoid situations if user registered IDeviceDisplay with the different lifetime (Transcient)

**Reply by durandt** [reply]
Hi @VladislavAntonyuk !

Thank you for your answer.

Sorry for being unclear, I am working on my own BottomSheet with limited features (see #947), but I will try to follow Community Toolkit's design patterns to make it easier to (possibly) contribute in the future.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889

**Reply by durandt** [reply]
@CliffAgius This was not a feature request. I was asking about what was the right way to use DI.

**Reply by CliffAgius** [reply]
Sorry I closed as it was an old discussion that had gone stale as discussed in the stand-up.  Are you still looking for help/advice if so I can re-open this.

**Reply by durandt** [reply]
@CliffAgius Ok cleaning up is always good :) That was no blocker and I solved this in some way so no problem.

**Reply by CliffAgius** [reply]
Ok in that case I will leave this closed and thanks for replying.
